From abedeba11521e29947e65fa1df02104647c448f7 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 30 Sep 2022 11:47:40 +0300
Subject: [PATCH] [SW-103485] gaudi3: errata wa for mem cpy wr os

As a workaround for hw errata GAUDI3_0315, memcpy wr_os register
is set to 62. Default value for the register is 32 . The workaround
would increase performance and make sure the register is not programmed
with the error case value (i.e 63)

Change-Id: I4015f132c6c5d3882ac12531a2aaffad1fcdc2d7
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 73a6185..9ccd306 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -51,6 +51,7 @@
 	return 0;
 }
 
+#define MEM_CPY_ERRATA_WR_OS_VAL		62	/* wr os should be 62 as per HW errata GAUDI3_0315 */
 int arcboot_rom_mem_init(uint64_t addr, uint32_t size)
 {
 	uint64_t start = arcboot_rom_get_ts_us();
@@ -67,6 +68,11 @@
 	mem_cfg |= (size & D0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_MEM_SIZE_M);
 	mem_cfg |= D0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_USE_CONST_M;
 
+	/*
+	 * WA for HW errata GAUDI3_0315
+	 */
+	mem_cfg |= (MEM_CPY_ERRATA_WR_OS_VAL << D0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_WR_OS_S);
+
 	regmap_writel(mem_cfg, mmD0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
 	regmap_writel(0x0, mmD0_PSOC_GLOBAL_CONF_MEM_CPY_CONST);
 
